includes/modules/attributes.php

```diff
@@ -540,7 +542,7 @@
 
 
         // default
-        // find default attribute if set for default dropdown
+        // find default attribute if set. Intended for dropdown's default.
         if ($products_options->fields['attributes_default'] == '1') {
             $selected_attribute = $products_options_value_id;

```

Comment:
Maybe could even use a "new variable here" and not have to effect other *outer* code:
````php
$selected_attribute = isset($selected_select_attribute) ? $selected_select_attribute : false; // $selected_select_attribute ?? false; // Possible only in PHP 7.x
if ($products_options->fields['attributes_default'] == '1') {
  $selected_attribute = $selected_select_attribute = $products_options_value_id;
}
````
with `$selected_select_attribute = false` in the outer loop?

(`isset` chosen instead of `!empty` because of the potential for `$products_options_values_id` being 0 as for files, text, and any other option type that has been developed...)

This would for all intensive purposes retain the dropdown's `$selected_attribute` value and not necessarily require a new variable to otherwise be considered outside the inner loop (though those that wish to know/retain the value while in the inner loop would need to pick up on that new variable).

Ughhh, yes, I understand the consternation of deciding/contemplating which way to go... Retain existing functionality with a cleaner set of code or revise how it works and what is needed to potentially take advantage of the original ZC 1.5.7 redesign...